Mac-App/Frontend/Frontend/Views/MainSplitView/MainSplitView.swift

```diff
@@ -9,12 +9,21 @@
 import SwiftUI
 
 public struct MainSplitView: View {
-    private var viewData: ObservableData<DependencyTree.State>
-    public init(viewData: ObservableData<DependencyTree.State>) {
-        self.viewData = viewData
+    private var dependencyTreeState: Observable<DependencyTree.State>
+    private var projectDetailsStatus: Observable<ProjectDetails.Status>
```

Comment:
A few notes here: 

- Here you can compare the before/after regarding naming (State/Status). So the intention is for views to get status injected so you switch over different statuses instead of state, which you don't really switch over but directly consume (like in success case for example)

- This is a design issue I don't like but not sure how to address. The more views are added to the project, the longer this initialiser will become. You can see how these objects get sent all the way down to their ultimate SwiftUI views. 
Sure, some views could reuse some of these, but most likely that won't be the usual case. I tried using `@EnvironmentObject` so that all subviews know about these dependencies allowing initialisers to be completely clean, but that does not work the moment you interact with `NSViewControllerRepresentable` because that's not interoperable with how SwiftUI  works.

Reply:
I think its alright for now, but I can imagine how crowded all the initialisers can get, maybe it would make sense to tackle this at a later time, or maybe spend some time on this particular problem.

Reply:
Yeah, this is actually the other design problem I told you about I was having 😅 

I think it's an interesting problem to solve.